File: RequestLoggerSpa/request-logger/src/app/models/mocked-response.ts
export class MockedResponse {
  constructor(
    route: string,
    method: string,
    statusCode: number,
    body: string | undefined = undefined,
    headers: any | undefined = undefined
  ) {
    if (!route || !method || !statusCode) {
      throw new Error('Route, method and status code must be provided');
    }
    this.route = route;
    this.method = method;
    this.statusCode = statusCode;
    this.body = body;
    this.headers = headers;
  }

  route: string = '';

  method: string = '';

  headers: any = {};

  body: string | undefined;

  statusCode: number = 200;

  addHeader(key: string, value: string) {
    this.headers[key] = value;
  }
}

File: RequestLoggerApi/RequestLogger.Domain/Entities/MockedResponse.cs
using System;
using System.Collections.Generic;
using System.Net;
using System.Net.Http;

namespace RequestLogger.Domain.Entities
{
    public class MockedResponse
    {
        private string _route;
        public string Route
        {
            get => _route;
            set
            {
                if (value == null)
                {
                    throw new ArgumentNullException(nameof(Route));
                }
                if (!Uri.IsWellFormedUriString(value, UriKind.Relative) || !value.StartsWith('/'))
                {
                    throw new ArgumentException($"Route <{value}> has an invalid format");
                }

                _route = value;
            }
        }

        public HttpMethod Method { get; set; }

        public IDictionary<string, string> Headers { get; set; } = new Dictionary<string, string>();

        public string Body { get; set; } = string.Empty;

        private HttpStatusCode _statusCode = HttpStatusCode.OK;
        public HttpStatusCode StatusCode
        {
            get => _statusCode;
            set
            {
                if (!Enum.IsDefined(typeof(HttpStatusCode), value))
                {
                    throw new ArgumentException($"Invalid status code : {value}");
                }

                _statusCode = value;
            }
        }
    }
}
